joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/Py2CpgOnFileSystem.scala

```diff
@@ -39,35 +50,56 @@ class Py2CpgOnFileSystem extends X2CpgFrontend[Py2CpgOnFileSystemConfig] {
     logConfiguration(config)
 
     X2Cpg.withNewEmptyCpg(config.outputPath, config) { (cpg, _) =>
-      val ignorePrefixes =
+      val venvIgnorePath =
         if (config.ignoreVenvDir) {
           config.venvDir :: Nil
         } else {
           Nil
         }
+      val inputPath         = Path.of(config.inputPath)
+      val ignoreDirNamesSet = config.ignoreDirNames.toSet
+      val absoluteIgnorePaths = (config.ignorePaths ++ venvIgnorePath).map { path =>
+        inputPath.resolve(path)
+      }
+
       val inputFiles = SourceFiles
         .determine(config.inputPath, Set(".py"), config)
         .map(x => Path.of(x))
-        .filterNot { file =>
-          val relativeFile = Path.of(config.inputPath).relativize(file)
-          ignorePrefixes.exists(prefix => relativeFile.startsWith(prefix))
+        .filter { file => filterIgnoreDirNames(file, inputPath, ignoreDirNamesSet) }
+        .filter { file =>
+          !absoluteIgnorePaths.exists(ignorePath => file.startsWith(ignorePath))
         }
+
       val inputProviders = inputFiles.map { inputFile => () =>
         {
           val content = IOUtils.readLinesInFile(inputFile).mkString("\n")
-          Py2Cpg.InputPair(content, Paths.get(config.inputPath).relativize(inputFile).toString)
+          Py2Cpg.InputPair(content, inputPath.relativize(inputFile).toString)
         }
       }
       val py2Cpg = new Py2Cpg(inputProviders, cpg, config.inputPath, config.requirementsTxt, config.schemaValidation)
       py2Cpg.buildCpg()
     }
   }
 
+  private def filterIgnoreDirNames(file: Path, inputPath: Path, ignoreDirNamesSet: Set[String]): Boolean = {
```

Comment:
the name makes this a bit hard to read...
it doesn't really filter anything. 
so how about `ignorePath` or so?